src/activities/Elsa.Activities.RabbitMq/Decorators/MessageHandlingTransportDecorator .cs

```diff
@@ -0,0 +1,33 @@
+using Rebus.Messages;
+using Rebus.Transport;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elsa.Activities.RabbitMq.Decorators
+{
+    public class MessageHandlingTransportDecorator : ITransport
+    {
+        static readonly Task<TransportMessage> NoMessage = Task.FromResult<TransportMessage>(null);
+
+        readonly ITransport _transport;
+        readonly MessageHandlingToggle _toggle;
+
+        public MessageHandlingTransportDecorator(ITransport transport, MessageHandlingToggle toggle)
+        {
+            _transport = transport;
+            _toggle = toggle;
+        }
+
+        public string Address => _transport.Address;
+
+        public void CreateQueue(string address) => _transport.CreateQueue(address);
+
+        public Task Send(string destinationAddress, TransportMessage message, ITransactionContext context)
+            => _transport.Send(destinationAddress, message, context);
+
+        public Task<TransportMessage> Receive(ITransactionContext context, CancellationToken cancellationToken)
+            => _toggle.IsReceivingMessages
```

Comment:
Can you explain the purpose of the toggle and why it is needed?

Reply:
The purpose of the toggle is to have more strict control on when activity is able to receive messages. The idea behind it, is that receiver activity is meant to handle a single message, but message handler can be triggered multiple times in a very short period, as soon as it is registered. Even though it doesn't seem to be affecting workflow in any way, messages from the broker are getting consumed by activity without any real need for it. Toggle is just a fast and easy way to flip a switch and prevent client from handling any more incoming messages.

Reply:
I see. Do we understand why the message handler is triggered multiple times?

Reply:
OK. But could it then not also happen that there are different messages enqueued that each need to be processed, but as long as this activity is executing, all subsequent messages are now being ignored?

If so, then perhaps a better approach might be to not discard messages, but actually handle them and let the engine ignore them. Unless you can be sure that a secondary message is in fact the same message that was already handled.

Reply:
I think I agree with @sfmskywalker . Let the system handle the messages. It will need to anyway as soon as the workflow gets published. At least if I understand the issue correctly.

Reply:
@sfmskywalker I've removed toggle in the last commit.